### PR TITLE
[RPC] dumpmasterprivkey command

### PR DESCRIPTION
**Pull request by achow101**
RPC command to export the master private key from a wallet.

### CODE REVIEW

**Comment by luke-jr** [comment]
`dumpwallet` already does this...?

**Comment by achow101** [comment]
`dumpwallet` requires that you dump to a file then find that file and open it to get the master private key. This just gives it to you with the command.

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -2602,6 +2603,7 @@ static const CRPCCommand commands[] =
     { "wallet",             "addwitnessaddress",        &addwitnessaddress,        true,   {"address"} },
     { "wallet",             "backupwallet",             &backupwallet,             true,   {"destination"} },
     { "wallet",             "dumpprivkey",              &dumpprivkey,              true,   {"address"}  },
+    { "wallet",             "dumpmasterprivkey",        &dumpmasterprivkey,         true,   {} },
```
Comment (jonasschnelli) [inline src/wallet/rpcwallet.cpp]: nit: column alignment.

File: src/wallet/rpcdump.cpp [hunk 2]
```diff
@@ -550,6 +550,48 @@ UniValue dumpprivkey(const JSONRPCRequest& request)
...
+            " \"key\"                 (string) The HD master private key\n"
+            "\nExamples:\n"
+            + HelpExampleCli("dumpmasterprivkey", "")
+        );
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    EnsureWalletIsUnlocked();
+
+    CKeyID masterKeyID = pwalletMain->GetHDChain().masterKeyID;
```
Comment (jonasschnelli) [inline src/wallet/rpcdump.cpp]: Use `CWallet::IsHDEnabled()`?

Comment (jonasschnelli) [inline src/wallet/rpcdump.cpp]: Well,... you need the masterKeys `CKeyID` anyway,... but I guess it would still make sense to use `CWallet::IsHDEnabled()`

**Review by jonasschnelli — CHANGES_REQUESTED** [review]
Not sure if we should add more private key exporting features because it increases the risk of doing something wrong with that.

If we are going to add this, it would require...
... add `dumpmasterprivkey` to the UI's sensitiv data filter
... add a test somewhere

**Comment by achow101** [comment]
@jonasschnelli there is no sensitive data filter for the `dump*` commands. The user is not entering sensitive data, only sensitive data being output.

**Comment by achow101** [comment]
Addressed nits and added a test.

It may need more tests, but I'm not sure what actually needs to be tested.

**Comment by jonasschnelli** [comment]
> @jonasschnelli there is no sensitive data filter for the dump* commands. The user is not entering sensitive data, only sensitive data being output.

Argh. Right. I was confusing the sensitiv history filter with the idea of having a filter that disables certain sensitive commands.

**Comment by achow101** [comment]
Closing this for now

**Pull request closed without merging**